components/automate-ui/src/app/modules/infra-proxy/infra-environment-constraint/infra-environment-constraint.component.ts

```diff
@@ -0,0 +1,188 @@
+import {
+  Component,
+  OnInit,
+  ViewChild,
+  Input,
+  Output,
+  EventEmitter
+} from '@angular/core';
+
+import { FormBuilder, Validators, FormGroup } from '@angular/forms';
+import { Regex } from 'app/helpers/auth/regex';
+
+import { MatSelect } from '@angular/material/select';
+import { DynamicGrid } from '../create-environment-modal/create-environment-modal.component';
+
+@Component({
+  selector: 'app-infra-environment-constraint',
+  templateUrl: './infra-environment-constraint.component.html',
+  styleUrls: ['./infra-environment-constraint.component.scss']
+})
+
+export class InfraEnvironmentConstraintComponent implements OnInit {
+
+  @Input() constraintKeys: string[] = [];
+  @Input() dynamicArray: Array<DynamicGrid> = [];
+  @Input() name_id: string;
+  @Output() constraintValues: EventEmitter<DynamicGrid[]> = new EventEmitter();
+  @ViewChild(MatSelect) select: MatSelect;
+
+  public conflictError = false;
+  public operator_id = '';
+  public operatorKeys: string[] = [];
+  public secondFormGroup: FormGroup;
+  public selectedLangs: string[] = [];
+  public rowKeys: string[] = [];
+
+  constructor(
+    private fb: FormBuilder
+  ) {
+    this.secondFormGroup = this.fb.group({
+      version: ['', [Validators.required, Validators.pattern(Regex.patterns.VALID_VERSION)]]
+    });
+
+    this.operatorKeys = ['~>', '>=', '>', '=', '<', '<='];
+
+    this.operator_id = this.operatorKeys[0];
+  }
+
+  ngOnInit() {
+
+    this.conflictError = false;
+    this.constraintKeys.forEach(element =>
+
+      this.rowKeys.push(element)
+    );
+  }
+
+  selected(value, i) {
```

Comment:
`i` is not an appropriate name for a function argument; needs to convey some meaning. Perhaps `languageIndex` ?

Reply:
added type in all the functions